### ISSUE TITLE
Add Bazel rules to extract Go compilations for testing. [phab:D1205]

### ISSUE THREAD

**Issue by kythe** [google/kythe#1579]
_Created by **fromberger** at 2016-12-29 22:34:31_:

**Comment by kythe** [comment]
_**shahms** commented inline (context lost during migration) at 2016-12-29 22:50:25_:

I'd find this more readable as:
```
cmds += [
  'mkdir -p ' + fullpath.rpslit('/', 1)[0],
  "ln -s ....",
]
```

**Comment by kythe** [comment]
_**shahms** commented inline (context lost during migration) at 2016-12-29 22:50:25_:

same here (use += rather than several appends)

**Comment by kythe** [comment]
_**shahms** commented inline (context lost during migration) at 2016-12-29 22:50:25_:

`cmds += ['ln -s ....' % (...) for src in srcs]`

**Comment by kythe** [comment]
_**shahms** wrote at 2016-12-29 22:50:25_:

minor nits on bzl conventions otherwise seems reasonable

**Comment by kythe** [comment]
_**fromberger** wrote at 2016-12-29 22:58:18_:

  - Use += instead of list append.

**Comment by kythe** [comment]
_**fromberger** wrote at 2016-12-29 22:58:39_:

Thanks!

**Comment by kythe** [comment]
_**fromberger** wrote at 2016-12-29 23:00:38_:

Oops, missed the other batch. Fixed now.

**Comment by kythe** [comment]
_**fromberger** wrote at 2016-12-29 23:01:01_:

  - Fix another cluster of append uses.

**Comment by kythe** [comment]
_**fromberger** closed via commit 6ceb9e8f218d2090d544d7cfea41d585835683ad at 2016-12-29 23:05:16_
